pkg/protocol/protocol_warp_sync.go

```diff
@@ -213,12 +225,9 @@ func (w *WarpSyncProtocol) ProcessResponse(commitmentID iotago.CommitmentID, blo
 						targetEngine.BlockGadget.SetAccepted(block)
 
 						block.Notarized().OnUpdate(func(_ bool, _ bool) {
-							// Wait for all blocks to be notarized before forcing the commitment of the slot.
-							if notarizedBlocks.Add(1) != totalBlocks {
-								return
+							if notarizedBlocks.Add(1) == totalBlocks {
```

Comment:
Does this work in the case where there's blocks without a future cone in a slot? ie when we revive the chain? I think we should force accept the blocks we didn't accept within `minCommittableAge` and then mark it as fully booked (or by force accepting those blocks it'd become fully booked automatically I guess).

Reply:
Blocks without a future cone can become naturally booked, and therefore the slot will become fully booked. This is because booked means that the past cone needs to exist, the future cone doesn't matter there.
On this specific line we check for all the blocks to be _notarized_, meaning accepted; and we force accept all the blocks we have received via warpsync on line 225.